### PR TITLE
feature/datasources

### PR DESCRIPTION
**Pull request by a-cherednikov**

### CODE REVIEW

**Comment by a-cherednikov** [comment]
will be removed:

packages/config/apidefs/universal-graph.yaml
packages/config/datasources/contentful.yaml
packages/config/datasources/covid.yaml
packages/config/gateway.yaml

File: tsconfig.json [hunk 1]
```diff
@@ -35,6 +35,15 @@
     },
     {
       "path": "packages/gateway"
+    },
+    {
+      "path": "packages/datasource"
```
Comment (valcker) [inline tsconfig.json:40]: Don't want to sound too pedantic, but shall we name it "datasources" and not "datasource"? :)

File: packages/datasource/package.json [hunk 2]
```diff
@@ -0,0 +1,12 @@
+{
```
Comment (valcker) [inline packages/datasource/package.json:1]: Do we need a top-level package.json if in any case we include the datasources explicitely?

File: packages/datasource/index.ts [hunk 3]
```diff
@@ -0,0 +1,4 @@
+import contentfulDefinitiond from './contentful/src/definitions.json';
```
Comment (valcker) [inline packages/datasource/index.ts:1]: Typo

File: packages/datasource/index.ts [hunk 4]
```diff
@@ -0,0 +1,4 @@
...
+import slackDefinitions from './slack/src/definitions.json';
+
+export default { ...contentfulDefinitiond, ...slackDefinitions };
```
Comment (valcker) [inline packages/datasource/index.ts:4]: Same typo

File: packages/datasource/slack/src/index.ts [hunk 5]
```diff
@@ -0,0 +1,14 @@
+import { GetMeshSourceOptions, YamlConfig } from '@graphql-mesh/types';
+const OpenAPIHandler = require('@graphql-mesh/openapi');
+
+const OPENAPI_SPECIFICATION: string = "https://api.slack.com/specs/openapi/v2/slack_web.json";
```
Comment (valcker) [inline packages/datasource/slack/src/index.ts:4]: Replace double quotes with single ones.

File: packages/datasource/slack/src/index.ts [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+
+export default class SlackHandler extends OpenAPIHandler {
+  constructor({ name, config, cache }: GetMeshSourceOptions<YamlConfig.OpenapiHandler>) {
+    config.source = OPENAPI_SPECIFICATION;
+
+    // @ts-ignore
+    config.schemaHeaders = { authorization: `Bearer ${config.token}`};
```
Comment (valcker) [inline packages/datasource/slack/src/index.ts:11]: Missing space after ...token} and before `. Probably Prettier is not enabled?

File: packages/datasource/contentful/src/index.ts [hunk 7]
```diff
@@ -0,0 +1,10 @@
+import { GetMeshSourceOptions, YamlConfig } from '@graphql-mesh/types';
+const GraphQLHandler = require('@graphql-mesh/graphql');
+
+export default class ContentfulHandler extends GraphQLHandler {
+  constructor({ name, config, cache }: GetMeshSourceOptions<YamlConfig.GraphQLHandler>) {
+    // @ts-ignore
+    config.schemaHeaders = { authorization: `Bearer ${config.token}`};
```
Comment (valcker) [inline packages/datasource/contentful/src/index.ts:7]: Some formatting is missing (trailing spaces after "`" and "}" ). Probably Prettier is disabled.

**Review by valcker — APPROVED** [review]

File: packages/datasource/contentful/src/index.ts (hunk 7 above)
Comment (a-cherednikov) [reply packages/datasource/contentful/src/index.ts:7]: +

File: packages/datasource/slack/src/index.ts (hunk 6 above)
Comment (a-cherednikov) [reply packages/datasource/slack/src/index.ts:11]: `+`

File: packages/datasource/slack/src/index.ts (hunk 5 above)
Comment (a-cherednikov) [reply packages/datasource/slack/src/index.ts:4]: `+`

File: packages/datasource/index.ts (hunk 4 above)
Comment (a-cherednikov) [reply packages/datasource/index.ts:4]: `+`

File: packages/datasource/index.ts (hunk 3 above)
Comment (a-cherednikov) [reply packages/datasource/index.ts:1]: `+`

File: packages/datasource/package.json (hunk 2 above)
Comment (a-cherednikov) [reply packages/datasource/package.json:1]: yes unfortunately

File: tsconfig.json (hunk 1 above)
Comment (a-cherednikov) [reply tsconfig.json:40]: `+`, renamed

**Pull request merged**